### ISSUE TITLE
[TW-958] Issues with cmake and make and lua under 1.9.4

### ISSUE THREAD

**Issue by tbabej** [tbabej/testimportfinal#930]
_Cory Donnelly on 2011-01-06T20:46:04Z says:_ 

 When attempting to (at)cmake .@ under osx 10.4.11 without lua installed, I receive the following:

<pre>
...
-- Configuring man pages
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
LUA_INCLUDE_DIR (ADVANCED)
   used as include directory in directory /Users/cory/git/task.git/src
   used as include directory in directory /Users/cory/git/task.git/test

-- Configuring incomplete, errors occurred!
</pre>

After installing lua, (at)cmake .@ completes successfully.  Running (at)make@ returns:

<pre>
[  2%] Building CXX object src/CMakeFiles/task.dir/API.cpp.o
/Users/cory/git/task.git/src/API.cpp: In function 'int api_task_lua_version(lua_State*)':
/Users/cory/git/task.git/src/API.cpp:75: error: 'LUA_RELEASE' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp: In function 'int api_task_get_due(lua_State*)':
/Users/cory/git/task.git/src/API.cpp:334: error: 'lua_pushinteger' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp: In function 'int api_task_get_entry(lua_State*)':
/Users/cory/git/task.git/src/API.cpp:351: error: 'lua_pushinteger' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp: In function 'int api_task_get_start(lua_State*)':
/Users/cory/git/task.git/src/API.cpp:368: error: 'lua_pushinteger' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp: In function 'int api_task_get_end(lua_State*)':
/Users/cory/git/task.git/src/API.cpp:385: error: 'lua_pushinteger' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp: In function 'int api_task_get_until(lua_State*)':
/Users/cory/git/task.git/src/API.cpp:410: error: 'lua_pushinteger' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp: In function 'int api_task_get_wait(lua_State*)':
/Users/cory/git/task.git/src/API.cpp:427: error: 'lua_pushinteger' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp: In member function 'void API::initialize()':
/Users/cory/git/task.git/src/API.cpp:528: error: 'luaL_openlibs' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp: In member function 'bool API::callProgramHook(const std::string&, const std::string&)':
/Users/cory/git/task.git/src/API.cpp:619: error: 'lua_tointeger' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp: In member function 'bool API::callTaskHook(const std::string&, const std::string&, Task&)':
/Users/cory/git/task.git/src/API.cpp:694: error: 'lua_tointeger' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp: In member function 'bool API::callFieldHook(const std::string&, const std::string&, const std::string&, std::string&)':
/Users/cory/git/task.git/src/API.cpp:748: error: 'lua_tointeger' was not declared in this scope
make[2]: *** [src/CMakeFiles/task.dir/API.cpp.o] Error 1
make[1]: *** [src/CMakeFiles/task.dir/all] Error 2
make: *** [all] Error 2
</pre>

**Comment by tbabej** [comment]
Migrated metadata:
```
Created: 2011-01-06T20:46:04Z
Modified: 2014-02-09T02:13:07Z```

**Comment by tbabej** [comment]
_Cory Donnelly on 2011-01-06T21:10:08Z says:_ 

 The above output was received with lua 5.0.2 installed -- with 5.1.4 installed, I received the following:

<pre>
$ make   
[  2%] Building CXX object src/CMakeFiles/task.dir/API.cpp.o
In file included from /Users/cory/git/task.git/src/Hooks.h:32,
                 from /Users/cory/git/task.git/src/Context.h:39,
                 from /Users/cory/git/task.git/src/API.cpp:52:
/Users/cory/git/task.git/src/API.h:39:19: error: lua.h: No such file or directory
/Users/cory/git/task.git/src/API.h:40:22: error: lualib.h: No such file or directory
/Users/cory/git/task.git/src/API.h:41:23: error: lauxlib.h: No such file or directory
/Users/cory/git/task.git/src/API.h:62: error: ISO C++ forbids declaration of 'lua_State' with no type
/Users/cory/git/task.git/src/API.h:62: error: expected ';' before '*' token
/Users/cory/git/task.git/src/API.cpp:63: error: 'lua_State' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:63: error: 'L' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:64: error: expected ',' or ';' before '{' token
/Users/cory/git/task.git/src/API.cpp:72: error: 'lua_State' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:72: error: 'L' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:73: error: expected ',' or ';' before '{' token
/Users/cory/git/task.git/src/API.cpp:82: error: 'lua_State' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:82: error: 'L' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:83: error: expected ',' or ';' before '{' token
/Users/cory/git/task.git/src/API.cpp:106: error: 'lua_State' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:106: error: 'L' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:107: error: expected ',' or ';' before '{' token
/Users/cory/git/task.git/src/API.cpp:135: error: 'lua_State' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:135: error: 'L' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:136: error: expected ',' or ';' before '{' token
/Users/cory/git/task.git/src/API.cpp:191: error: 'lua_State' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:191: error: 'L' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:192: error: expected ',' or ';' before '{' token
/Users/cory/git/task.git/src/API.cpp:199: error: 'lua_State' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:199: error: 'L' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:200: error: expected ',' or ';' before '{' token
/Users/cory/git/task.git/src/API.cpp:207: error: 'lua_State' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:207: error: 'L' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:208: error: expected ',' or ';' before '{' token
/Users/cory/git/task.git/src/API.cpp:217: error: 'lua_State' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:217: error: 'L' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:218: error: expected ',' or ';' before '{' token
/Users/cory/git/task.git/src/API.cpp:256: error: 'lua_State' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:256: error: 'L' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:257: error: expected ',' or ';' before '{' token
/Users/cory/git/task.git/src/API.cpp:267: error: 'lua_State' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:267: error: 'L' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:268: error: expected ',' or ';' before '{' token
/Users/cory/git/task.git/src/API.cpp:286: error: 'lua_State' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:286: error: 'L' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:287: error: expected ',' or ';' before '{' token
/Users/cory/git/task.git/src/API.cpp:297: error: 'lua_State' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:297: error: 'L' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:298: error: expected ',' or ';' before '{' token
/Users/cory/git/task.git/src/API.cpp:316: error: 'lua_State' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:316: error: 'L' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:317: error: expected ',' or ';' before '{' token
/Users/cory/git/task.git/src/API.cpp:327: error: 'lua_State' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:327: error: 'L' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:328: error: expected ',' or ';' before '{' token
/Users/cory/git/task.git/src/API.cpp:344: error: 'lua_State' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:344: error: 'L' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:345: error: expected ',' or ';' before '{' token
/Users/cory/git/task.git/src/API.cpp:361: error: 'lua_State' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:361: error: 'L' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:362: error: expected ',' or ';' before '{' token
/Users/cory/git/task.git/src/API.cpp:378: error: 'lua_State' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:378: error: 'L' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:379: error: expected ',' or ';' before '{' token
/Users/cory/git/task.git/src/API.cpp:403: error: 'lua_State' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:403: error: 'L' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:404: error: expected ',' or ';' before '{' token
/Users/cory/git/task.git/src/API.cpp:420: error: 'lua_State' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:420: error: 'L' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:421: error: expected ',' or ';' before '{' token
/Users/cory/git/task.git/src/API.cpp: In constructor 'API::API()':
/Users/cory/git/task.git/src/API.cpp:509: error: class 'API' does not have any field named 'L'
/Users/cory/git/task.git/src/API.cpp: In destructor 'API::~API()':
/Users/cory/git/task.git/src/API.cpp:516: error: 'L' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:518: error: 'lua_close' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp: In member function 'void API::initialize()':
/Users/cory/git/task.git/src/API.cpp:527: error: 'L' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:527: error: 'lua_open' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:528: error: 'luaL_openlibs' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:531: error: 'lua_pushcfunction' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:531: error: 'lua_setglobal' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp: In member function 'bool API::callProgramHook(const std::string&, const std::string&)':
/Users/cory/git/task.git/src/API.cpp:601: error: 'L' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:601: error: 'lua_getglobal' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:602: error: 'lua_isfunction' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:604: error: 'lua_pop' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:609: error: 'lua_pcall' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:610: error: 'lua_tostring' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:613: error: 'lua_isnumber' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:616: error: 'lua_isstring' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:616: error: 'lua_isnil' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:619: error: 'lua_tointeger' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:620: error: 'lua_tostring' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:633: error: 'lua_pop' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp: In member function 'bool API::callTaskHook(const std::string&, const std::string&, Task&)':
/Users/cory/git/task.git/src/API.cpp:667: error: 'L' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:667: error: 'lua_getglobal' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:668: error: 'lua_isfunction' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:670: error: 'lua_pop' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:675: error: 'lua_pushnumber' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:681: error: 'lua_pcall' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:682: error: 'lua_tostring' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:688: error: 'lua_isnumber' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:691: error: 'lua_isstring' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:691: error: 'lua_isnil' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:694: error: 'lua_tointeger' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:695: error: 'lua_tostring' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:708: error: 'lua_pop' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp: In member function 'bool API::callFieldHook(const std::string&, const std::string&, const std::string&, std::string&)':
/Users/cory/git/task.git/src/API.cpp:722: error: 'L' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:722: error: 'lua_getglobal' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:723: error: 'lua_isfunction' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:725: error: 'lua_pop' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:730: error: 'lua_pushstring' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:734: error: 'lua_pcall' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:735: error: 'lua_tostring' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:738: error: 'lua_isstring' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:741: error: 'lua_isnumber' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:744: error: 'lua_isstring' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:744: error: 'lua_isnil' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:747: error: 'lua_tostring' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:748: error: 'lua_tointeger' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:765: error: 'lua_pop' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp: In member function 'void API::loadFile(const std::string&)':
/Users/cory/git/task.git/src/API.cpp:776: error: 'L' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:776: error: 'luaL_loadfile' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:776: error: 'lua_pcall' was not declared in this scope
/Users/cory/git/task.git/src/API.cpp:777: error: 'lua_tostring' was not declared in this scope
make[2]: *** [src/CMakeFiles/task.dir/API.cpp.o] Error 1
make[1]: *** [src/CMakeFiles/task.dir/all] Error 2
make: *** [all] Error 2
</pre>

**Comment by tbabej** [comment]
_Federico Hernandez on 2011-01-07T05:38:44Z says:_ 

 Cory,

I am not at the computer until this evening. But,

* did you re-run cmake after installing the lua version?
* what are the values for the corresponding lua variables in the cmakecache file?
* where are the lua include files and library on your system? could you see if you find the files make is complaining about on your system?

Thx for beta-testing our cmake efforts and discovering issues.

**Comment by tbabej** [comment]
_Cory Donnelly on 2011-01-07T09:18:44Z says:_ 

 I figured out what was going on -- in addition to lua51 (and possibly lua51-shlibs, I didn't try it without it) it was necessary to install lua51-dev. Otherwise, the include files and library didn't get installed to /sw/include by fink.

We should (sorry if we've already done this, I've been out of the loop for a few weeks):

- Identify and document the minimum version of lua required
- Tweak cmake to look for those versions
- Fail gracefully if lua isn't present

**Comment by tbabej** [comment]
_Federico Hernandez on 2011-01-07T10:45:22Z says:_ 

 You are right about the 3 points. It is on the list. The cmake integration is work in progress. You pulled pre-alpha software.

Cmake is already looking for 5.1 version of lua - using the find_package macro. It will be documented once cmake is done.
